src/jdk.sctp/unix/native/libsctp/SctpChannelImpl.c

```diff
@@ -243,12 +246,14 @@ void handleSendFailed
 
         if (remaining > 0) {
             if ((rv = recvmsg(fd, msg, 0)) < 0) {
+                free(addressP);
                 handleSocketError(env, errno);
                 return;
```

Comment:
At this point the direct ByteBuffer has been successfully allocated and refers to the memory at addressP. But the direct ByteBuffer, while in the Java heap, will not have any references to it, so freeing addressP should be fine here.